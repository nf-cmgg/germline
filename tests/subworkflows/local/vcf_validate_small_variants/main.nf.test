nextflow_workflow {

    name "Test Workflow VCF_VALIDATE_SMALL_VARIANTS"
    script "subworkflows/local/vcf_validate_small_variants/main.nf"
    workflow "VCF_VALIDATE_SMALL_VARIANTS"

    tag "subworkflows"
    tag "subworkflow_local"
    tag "vcf_validate_small_variants"

    test("vcf_validate_small_variants - default") {

        setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
                process {
                    """
                    input[0] = Channel.value([
                        [id:"sdf"],
                        file(params.sdf, checkIfExists:true)
                    ])
                    """
                }
            }
        }

        when {
            params {
                validate = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_count:3, caller:"haplotypecaller"],
                    file(params.famvcf, checkIfExists:true),
                    file(params.famtbi, checkIfExists:true),
                    file(params.vcf1, checkIfExists:true),
                    file(params.tbi1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_count:3, caller:"haplotypecaller"],
                    file(params.bed, checkIfExists:true),
                    []
                ])
                input[2] = [[],[]]
                input[3] = [[],[]]
                input[4] = UNTAR.out.untar
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = [[],[]]
                input[8] = "vcfeval"
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.happy_vcf,
                    workflow.out.happy_tbi,
                    workflow.out.happy_indel_roc,
                    workflow.out.happy_indel_roc_pass,
                    workflow.out.happy_snp_roc,
                    workflow.out.happy_snp_roc_pass,
                    workflow.out.happy_roc,
                    workflow.out.happy_summary,
                    workflow.out.happy_extended_csv,
                    workflow.out.vcfeval_true_positive_vcf,
                    workflow.out.vcfeval_true_positive_vcf_tbi,
                    workflow.out.vcfeval_false_negative_vcf,
                    workflow.out.vcfeval_false_negative_vcf_tbi,
                    workflow.out.vcfeval_false_positive_vcf,
                    workflow.out.vcfeval_false_positive_vcf_tbi,
                    workflow.out.vcfeval_true_positive_baseline_vcf,
                    workflow.out.vcfeval_true_positive_baseline_vcf_tbi,
                    workflow.out.vcfeval_summary,
                    workflow.out.vcfeval_phasing,
                    workflow.out.vcfeval_snp_roc,
                    workflow.out.vcfeval_non_snp_roc,
                    workflow.out.vcfeval_weighted_roc,
                    workflow.out.rtgtools_snp_png_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                    workflow.out.rtgtools_non_snp_png_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                    workflow.out.rtgtools_weighted_png_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                    workflow.out.rtgtools_snp_svg_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                    workflow.out.rtgtools_non_snp_svg_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                    workflow.out.rtgtools_weighted_svg_rocplot.collect { it.collect { it instanceof Map ? it : file(it).name } },
                ).match("default") }
            )
        }

    }

}
