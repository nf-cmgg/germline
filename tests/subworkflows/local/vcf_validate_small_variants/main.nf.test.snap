{
    "default": {
        "content": [
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.tp.vcf.gz:md5,b28d3e84a6efa95bfbf673de310f62c1"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.tp.vcf.gz.tbi:md5,8f8356fd93e093e9deb1c2c4c72cb65e"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.fn.vcf.gz:md5,99830701ae0191110b706b20abe56857"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.fn.vcf.gz.tbi:md5,092a7a3162e7cff25d273525751eb284"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.fp.vcf.gz:md5,2769bf65afefc821fe54a8b314b5d2be"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.fp.vcf.gz.tbi:md5,092a7a3162e7cff25d273525751eb284"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.tp-baseline.vcf.gz:md5,6de63c7e2fa458ca88ea78c36f579056"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.tp-baseline.vcf.gz.tbi:md5,8d399004d8c45fa795b70da32e23a6d5"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.summary.txt:md5,11416903ca828b22bd64cf297f2db999"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.phasing.txt:md5,38920536b8c3e241e873c07ba61762e6"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.snp_roc.tsv.gz:md5,d29e7796d37dada43378e466af98c35a"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.non_snp_roc.tsv.gz:md5,121b8b67d40efcaf1aad5064c6f3bfe9"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller"
                    },
                    "NA24143.haplotypecaller.weighted_roc.tsv.gz:md5,f016f9a10e793836b9b53d702e591454"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "snp"
                    },
                    "NA24143.haplotypecaller.snp.png"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "non_snp"
                    },
                    "NA24143.haplotypecaller.non_snp.png"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "weighted"
                    },
                    "NA24143.haplotypecaller.weighted.png"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "snp"
                    },
                    "NA24143.haplotypecaller.snp.svg"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "non_snp"
                    },
                    "NA24143.haplotypecaller.non_snp.svg"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "roc_type": "weighted"
                    },
                    "NA24143.haplotypecaller.weighted.svg"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T09:08:30.296761651"
    }
}